From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: hayanesuru <hayanesuru@outlook.jp>
Date: Fri, 23 May 2025 12:01:42 +0900
Subject: [PATCH] Cache block state tags


diff --git a/net/minecraft/server/ReloadableServerResources.java b/net/minecraft/server/ReloadableServerResources.java
index 42326fc..a69819a 100644
--- a/net/minecraft/server/ReloadableServerResources.java
+++ b/net/minecraft/server/ReloadableServerResources.java
@@ -109,5 +109,6 @@ public class ReloadableServerResources {
 
     public void updateStaticRegistryTags() {
         this.postponedTags.forEach(Registry.PendingTags::apply);
+        net.minecraft.world.level.block.Blocks.initPathType(); // Leaf - Cache block state tags
     }
 }
diff --git a/net/minecraft/world/level/block/Blocks.java b/net/minecraft/world/level/block/Blocks.java
index 57aad04..b7aa329 100644
--- a/net/minecraft/world/level/block/Blocks.java
+++ b/net/minecraft/world/level/block/Blocks.java
@@ -7066,4 +7066,14 @@ public class Blocks {
             }
         }
     }
+
+    // Leaf start - Cache block state tags
+    public static void initPathType() {
+        for (Block block : BuiltInRegistries.BLOCK) {
+            for (BlockState blockState : block.getStateDefinition().getPossibleStates()) {
+                blockState.initPathType();
+            }
+        }
+    }
+    // Leaf end - Cache block state tags
 }
diff --git a/net/minecraft/world/level/block/state/BlockBehaviour.java b/net/minecraft/world/level/block/state/BlockBehaviour.java
index f9ff9a3..503420b 100644
--- a/net/minecraft/world/level/block/state/BlockBehaviour.java
+++ b/net/minecraft/world/level/block/state/BlockBehaviour.java
@@ -471,6 +471,8 @@ public abstract class BlockBehaviour implements FeatureElement {
         private boolean emptyCollisionShape;
         private boolean emptyConstantCollisionShape;
         private VoxelShape constantCollisionShape;
+        public byte pathType; // Leaf - Cache block state tags
+        public int tagFlag; // Leaf - Cache block state tags
 
         private static void initCaches(final VoxelShape shape, final boolean neighbours) {
             ((ca.spottedleaf.moonrise.patches.collisions.shape.CollisionVoxelShape)shape).moonrise$isFullBlock();
@@ -638,6 +640,13 @@ public abstract class BlockBehaviour implements FeatureElement {
             // Paper end - optimise collisions
         }
 
+        // Leaf start - Cache block state tags
+        public void initPathType() {
+            pathType = (byte) net.minecraft.world.level.pathfinder.WalkNodeEvaluator.getPathTypeFromState(this.asState()).ordinal();
+            tagFlag = org.dreeam.leaf.util.BlockMasks.init(asState());
+        }
+        // Leaf end - Cache block state tags
+
         public Block getBlock() {
             return this.owner;
         }
diff --git a/net/minecraft/world/level/pathfinder/PathTypeCache.java b/net/minecraft/world/level/pathfinder/PathTypeCache.java
index 3b190ba..f745b26 100644
--- a/net/minecraft/world/level/pathfinder/PathTypeCache.java
+++ b/net/minecraft/world/level/pathfinder/PathTypeCache.java
@@ -24,7 +24,7 @@ public class PathTypeCache {
     }
 
     private PathType compute(BlockGetter level, BlockPos pos, int index, long packedPos) {
-        PathType pathTypeFromState = WalkNodeEvaluator.getPathTypeFromState(level, pos);
+        PathType pathTypeFromState = WalkNodeEvaluator.leafPathType(level, pos); // Leaf - Cache block state tags
         this.positions[index] = packedPos;
         this.pathTypes[index] = pathTypeFromState;
         return pathTypeFromState;
diff --git a/net/minecraft/world/level/pathfinder/PathfindingContext.java b/net/minecraft/world/level/pathfinder/PathfindingContext.java
index 4eca1dd..7f8b00f 100644
--- a/net/minecraft/world/level/pathfinder/PathfindingContext.java
+++ b/net/minecraft/world/level/pathfinder/PathfindingContext.java
@@ -27,7 +27,7 @@ public class PathfindingContext {
 
     public PathType getPathTypeFromState(int x, int y, int z) {
         BlockPos blockPos = this.mutablePos.set(x, y, z);
-        return this.cache == null ? WalkNodeEvaluator.getPathTypeFromState(this.level, blockPos) : this.cache.getOrCompute(this.level, blockPos);
+        return this.cache == null ? WalkNodeEvaluator.leafPathType(this.level, blockPos) : this.cache.getOrCompute(this.level, blockPos); // Leaf - Cache block state tags
     }
 
     public BlockState getBlockState(BlockPos pos) {
diff --git a/net/minecraft/world/level/pathfinder/WalkNodeEvaluator.java b/net/minecraft/world/level/pathfinder/WalkNodeEvaluator.java
index db6baaa..7eb7a8a 100644
--- a/net/minecraft/world/level/pathfinder/WalkNodeEvaluator.java
+++ b/net/minecraft/world/level/pathfinder/WalkNodeEvaluator.java
@@ -479,6 +479,17 @@ public class WalkNodeEvaluator extends NodeEvaluator {
         return pathType;
     }
 
+    // Leaf start - Cache block state tags
+    private static final PathType[] PATH_TYPES = PathType.values();
+    public static PathType leafPathType(BlockGetter level, BlockPos blockPos) {
+        BlockState blockState = level.getBlockStateIfLoaded(blockPos);
+        if (blockState == null) {
+            return PathType.BLOCKED;
+        }
+        return PATH_TYPES[blockState.pathType];
+    }
+    // Leaf end - Cache block state tags
+
     protected static PathType getPathTypeFromState(BlockGetter level, BlockPos pos) {
         // Paper start - Do not load chunks during pathfinding
         BlockState blockState = level.getBlockStateIfLoaded(pos);
@@ -486,6 +497,12 @@ public class WalkNodeEvaluator extends NodeEvaluator {
             return PathType.BLOCKED;
         }
         // Paper end
+        // Leaf start - Cache block state tags
+        return getPathTypeFromState(blockState);
+    }
+
+    public static PathType getPathTypeFromState(BlockState blockState) {
+        // Leaf end - Cache block state tags
         Block block = blockState.getBlock();
         if (blockState.isAir()) {
             return PathType.OPEN;
