From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: hayanesuru <hayanesuru@outlook.jp>
Date: Wed, 13 Aug 2025 23:35:42 +0900
Subject: [PATCH] fix purpur attribute base patch


diff --git a/net/minecraft/world/entity/GlowSquid.java b/net/minecraft/world/entity/GlowSquid.java
index 86a694b..bb95544 100644
--- a/net/minecraft/world/entity/GlowSquid.java
+++ b/net/minecraft/world/entity/GlowSquid.java
@@ -50,7 +50,7 @@ public class GlowSquid extends Squid {
     // Purpur start - Configurable entity base attributes
     @Override
     public void initAttributes() {
-        this.getAttribute(net.minecraft.world.entity.ai.attributes.Attributes.MAX_HEALTH).setBaseValue(this.level().purpurConfig.glowSquidMaxHealth);
+        this.setAttributeBaseValue(net.minecraft.world.entity.ai.attributes.Attributes.MAX_HEALTH, this.level().purpurConfig.glowSquidMaxHealth);
     }
     // Purpur end - Configurable entity base attributes
 
diff --git a/net/minecraft/world/entity/LivingEntity.java b/net/minecraft/world/entity/LivingEntity.java
index cbe9949..27fc48b 100644
--- a/net/minecraft/world/entity/LivingEntity.java
+++ b/net/minecraft/world/entity/LivingEntity.java
@@ -2857,6 +2857,16 @@ public abstract class LivingEntity extends Entity implements Attackable, Waypoin
         return this.getAttributes().getBaseValue(attribute);
     }
 
+    // Leaf start
+    public double setAttributeBaseValue(Holder<Attribute> attribute, double d) {
+        double old = this.getAttributes().getBaseValue(attribute);
+        if (old != d) {
+            this.getAttributes().getInstance(attribute).setBaseValue(d);
+        }
+        return old;
+    }
+    // Leaf end
+
     public AttributeMap getAttributes() {
         return this.attributes;
     }
diff --git a/net/minecraft/world/entity/ambient/Bat.java b/net/minecraft/world/entity/ambient/Bat.java
index c4985dd..e2ea4af 100644
--- a/net/minecraft/world/entity/ambient/Bat.java
+++ b/net/minecraft/world/entity/ambient/Bat.java
@@ -99,15 +99,15 @@ public class Bat extends AmbientCreature {
     // Purpur start - Configurable entity base attributes
     @Override
     public void initAttributes() {
-        this.getAttribute(Attributes.MAX_HEALTH).setBaseValue(this.level().purpurConfig.batMaxHealth);
-        this.getAttribute(Attributes.SCALE).setBaseValue(this.level().purpurConfig.batScale);
-        this.getAttribute(Attributes.FOLLOW_RANGE).setBaseValue(this.level().purpurConfig.batFollowRange);
-        this.getAttribute(Attributes.KNOCKBACK_RESISTANCE).setBaseValue(this.level().purpurConfig.batKnockbackResistance);
-        this.getAttribute(Attributes.MOVEMENT_SPEED).setBaseValue(this.level().purpurConfig.batMovementSpeed);
-        this.getAttribute(Attributes.FLYING_SPEED).setBaseValue(this.level().purpurConfig.batFlyingSpeed);
-        this.getAttribute(Attributes.ARMOR).setBaseValue(this.level().purpurConfig.batArmor);
-        this.getAttribute(Attributes.ARMOR_TOUGHNESS).setBaseValue(this.level().purpurConfig.batArmorToughness);
-        this.getAttribute(Attributes.ATTACK_KNOCKBACK).setBaseValue(this.level().purpurConfig.batAttackKnockback);
+        this.setAttributeBaseValue(Attributes.MAX_HEALTH, this.level().purpurConfig.batMaxHealth);
+        this.setAttributeBaseValue(Attributes.SCALE, this.level().purpurConfig.batScale);
+        this.setAttributeBaseValue(Attributes.FOLLOW_RANGE, this.level().purpurConfig.batFollowRange);
+        this.setAttributeBaseValue(Attributes.KNOCKBACK_RESISTANCE, this.level().purpurConfig.batKnockbackResistance);
+        this.setAttributeBaseValue(Attributes.MOVEMENT_SPEED, this.level().purpurConfig.batMovementSpeed);
+        this.setAttributeBaseValue(Attributes.FLYING_SPEED, this.level().purpurConfig.batFlyingSpeed);
+        this.setAttributeBaseValue(Attributes.ARMOR, this.level().purpurConfig.batArmor);
+        this.setAttributeBaseValue(Attributes.ARMOR_TOUGHNESS, this.level().purpurConfig.batArmorToughness);
+        this.setAttributeBaseValue(Attributes.ATTACK_KNOCKBACK, this.level().purpurConfig.batAttackKnockback);
     }
     // Purpur end - Configurable entity base attributes
 
diff --git a/net/minecraft/world/entity/animal/Bee.java b/net/minecraft/world/entity/animal/Bee.java
index 4821ec2..eadc64f 100644
--- a/net/minecraft/world/entity/animal/Bee.java
+++ b/net/minecraft/world/entity/animal/Bee.java
@@ -476,8 +476,8 @@ public class Bee extends Animal implements NeutralMob, FlyingAnimal {
     // Purpur start - Configurable entity base attributes
     @Override
     public void initAttributes() {
-        this.getAttribute(Attributes.MAX_HEALTH).setBaseValue(this.level().purpurConfig.beeMaxHealth);
-        this.getAttribute(Attributes.SCALE).setBaseValue(this.level().purpurConfig.beeScale);
+        this.setAttributeBaseValue(Attributes.MAX_HEALTH, this.level().purpurConfig.beeMaxHealth);
+        this.setAttributeBaseValue(Attributes.SCALE, this.level().purpurConfig.beeScale);
     }
     // Purpur end - Configurable entity base attributes
 
diff --git a/net/minecraft/world/entity/animal/Cat.java b/net/minecraft/world/entity/animal/Cat.java
index 09d518a..14dfac1 100644
--- a/net/minecraft/world/entity/animal/Cat.java
+++ b/net/minecraft/world/entity/animal/Cat.java
@@ -120,8 +120,8 @@ public class Cat extends TamableAnimal {
     // Purpur start - Configurable entity base attributes
     @Override
     public void initAttributes() {
-        this.getAttribute(Attributes.MAX_HEALTH).setBaseValue(this.level().purpurConfig.catMaxHealth);
-        this.getAttribute(Attributes.SCALE).setBaseValue(this.level().purpurConfig.catScale);
+        this.setAttributeBaseValue(Attributes.MAX_HEALTH, this.level().purpurConfig.catMaxHealth);
+        this.setAttributeBaseValue(Attributes.SCALE, this.level().purpurConfig.catScale);
     }
     // Purpur end - Configurable entity base attributes
 
diff --git a/net/minecraft/world/entity/animal/Chicken.java b/net/minecraft/world/entity/animal/Chicken.java
index eaf9138..f908d57 100644
--- a/net/minecraft/world/entity/animal/Chicken.java
+++ b/net/minecraft/world/entity/animal/Chicken.java
@@ -92,11 +92,11 @@ public class Chicken extends Animal {
     // Purpur start - Configurable entity base attributes
     @Override
     public void initAttributes() {
-        this.getAttribute(Attributes.MAX_HEALTH).setBaseValue(this.level().purpurConfig.chickenMaxHealth);
-        this.getAttribute(Attributes.SCALE).setBaseValue(this.level().purpurConfig.chickenScale);
+        this.setAttributeBaseValue(Attributes.MAX_HEALTH, this.level().purpurConfig.chickenMaxHealth);
+        this.setAttributeBaseValue(Attributes.SCALE, this.level().purpurConfig.chickenScale);
         // Purpur start - Chickens can retaliate
         if (level().purpurConfig.chickenRetaliate) {
-            this.getAttribute(Attributes.ATTACK_DAMAGE).setBaseValue(2.0D);
+            this.setAttributeBaseValue(Attributes.ATTACK_DAMAGE, 2.0D);
         }
         // Purpur end - Chickens can retaliate
     }
diff --git a/net/minecraft/world/entity/animal/Cod.java b/net/minecraft/world/entity/animal/Cod.java
index 7e08573..b954aa5 100644
--- a/net/minecraft/world/entity/animal/Cod.java
+++ b/net/minecraft/world/entity/animal/Cod.java
@@ -28,7 +28,7 @@ public class Cod extends AbstractSchoolingFish {
     // Purpur start - Configurable entity base attributes
     @Override
     public void initAttributes() {
-        this.getAttribute(net.minecraft.world.entity.ai.attributes.Attributes.MAX_HEALTH).setBaseValue(this.level().purpurConfig.codMaxHealth);
+        this.setAttributeBaseValue(net.minecraft.world.entity.ai.attributes.Attributes.MAX_HEALTH, this.level().purpurConfig.codMaxHealth);
     }
     // Purpur end - Configurable entity base attributes
 
diff --git a/net/minecraft/world/entity/animal/Cow.java b/net/minecraft/world/entity/animal/Cow.java
index b2fa170..aaa710c 100644
--- a/net/minecraft/world/entity/animal/Cow.java
+++ b/net/minecraft/world/entity/animal/Cow.java
@@ -51,9 +51,9 @@ public class Cow extends AbstractCow {
     // Purpur start - Configurable entity base attributes
     @Override
     public void initAttributes() {
-        this.getAttribute(net.minecraft.world.entity.ai.attributes.Attributes.MAX_HEALTH).setBaseValue(this.level().purpurConfig.cowMaxHealth);
-        this.getAttribute(net.minecraft.world.entity.ai.attributes.Attributes.SCALE).setBaseValue(this.level().purpurConfig.cowScale);
-        this.getAttribute(net.minecraft.world.entity.ai.attributes.Attributes.ATTACK_DAMAGE).setBaseValue(this.level().purpurConfig.cowNaturallyAggressiveToPlayersDamage); // Purpur - Cows naturally aggressive to players chance
+        this.setAttributeBaseValue(net.minecraft.world.entity.ai.attributes.Attributes.MAX_HEALTH, this.level().purpurConfig.cowMaxHealth);
+        this.setAttributeBaseValue(net.minecraft.world.entity.ai.attributes.Attributes.SCALE, this.level().purpurConfig.cowScale);
+        this.setAttributeBaseValue(net.minecraft.world.entity.ai.attributes.Attributes.ATTACK_DAMAGE, this.level().purpurConfig.cowNaturallyAggressiveToPlayersDamage); // Purpur - Cows naturally aggressive to players chance
     }
     // Purpur end - Configurable entity base attributes
 
diff --git a/net/minecraft/world/entity/animal/Dolphin.java b/net/minecraft/world/entity/animal/Dolphin.java
index 23696a5..cb53964 100644
--- a/net/minecraft/world/entity/animal/Dolphin.java
+++ b/net/minecraft/world/entity/animal/Dolphin.java
@@ -154,8 +154,8 @@ public class Dolphin extends AgeableWaterCreature {
     // Purpur start - Configurable entity base attributes
     @Override
     public void initAttributes() {
-        this.getAttribute(Attributes.MAX_HEALTH).setBaseValue(this.level().purpurConfig.dolphinMaxHealth);
-        this.getAttribute(Attributes.SCALE).setBaseValue(this.level().purpurConfig.dolphinScale);
+        this.setAttributeBaseValue(Attributes.MAX_HEALTH, this.level().purpurConfig.dolphinMaxHealth);
+        this.setAttributeBaseValue(Attributes.SCALE, this.level().purpurConfig.dolphinScale);
     }
     // Purpur end - Configurable entity base attributes
 
diff --git a/net/minecraft/world/entity/animal/Fox.java b/net/minecraft/world/entity/animal/Fox.java
index 8964bb5..619e7ec 100644
--- a/net/minecraft/world/entity/animal/Fox.java
+++ b/net/minecraft/world/entity/animal/Fox.java
@@ -183,8 +183,8 @@ public class Fox extends Animal {
     // Purpur start - Configurable entity base attributes
     @Override
     public void initAttributes() {
-        this.getAttribute(Attributes.MAX_HEALTH).setBaseValue(this.level().purpurConfig.foxMaxHealth);
-        this.getAttribute(Attributes.SCALE).setBaseValue(this.level().purpurConfig.foxScale);
+        this.setAttributeBaseValue(Attributes.MAX_HEALTH, this.level().purpurConfig.foxMaxHealth);
+        this.setAttributeBaseValue(Attributes.SCALE, this.level().purpurConfig.foxScale);
     }
     // Purpur end - Configurable entity base attributes
 
diff --git a/net/minecraft/world/entity/animal/HappyGhast.java b/net/minecraft/world/entity/animal/HappyGhast.java
index b7306bc..788ee26 100644
--- a/net/minecraft/world/entity/animal/HappyGhast.java
+++ b/net/minecraft/world/entity/animal/HappyGhast.java
@@ -134,13 +134,13 @@ public class HappyGhast extends Animal {
     // Purpur start - Configurable entity base attributes
     @Override
     public void initAttributes() {
-        this.getAttribute(Attributes.MAX_HEALTH).setBaseValue(this.level().purpurConfig.happyGhastMaxHealth);
-        this.getAttribute(Attributes.SCALE).setBaseValue(this.level().purpurConfig.happyGhastScale);
-        this.getAttribute(Attributes.TEMPT_RANGE).setBaseValue(this.level().purpurConfig.happyGhastTemptRange);
-        this.getAttribute(Attributes.FLYING_SPEED).setBaseValue(this.level().purpurConfig.happyGhastFlyingSpeed);
-        this.getAttribute(Attributes.MOVEMENT_SPEED).setBaseValue(this.level().purpurConfig.happyGhastMovementSpeed);
-        this.getAttribute(Attributes.FOLLOW_RANGE).setBaseValue(this.level().purpurConfig.happyGhastFollowRange);
-        this.getAttribute(Attributes.CAMERA_DISTANCE).setBaseValue(this.level().purpurConfig.happyGhastCameraDistance);
+        this.setAttributeBaseValue(Attributes.MAX_HEALTH, this.level().purpurConfig.happyGhastMaxHealth);
+        this.setAttributeBaseValue(Attributes.SCALE, this.level().purpurConfig.happyGhastScale);
+        this.setAttributeBaseValue(Attributes.TEMPT_RANGE, this.level().purpurConfig.happyGhastTemptRange);
+        this.setAttributeBaseValue(Attributes.FLYING_SPEED, this.level().purpurConfig.happyGhastFlyingSpeed);
+        this.setAttributeBaseValue(Attributes.MOVEMENT_SPEED, this.level().purpurConfig.happyGhastMovementSpeed);
+        this.setAttributeBaseValue(Attributes.FOLLOW_RANGE, this.level().purpurConfig.happyGhastFollowRange);
+        this.setAttributeBaseValue(Attributes.CAMERA_DISTANCE, this.level().purpurConfig.happyGhastCameraDistance);
     }
     // Purpur end - Configurable entity base attributes
 
diff --git a/net/minecraft/world/entity/animal/IronGolem.java b/net/minecraft/world/entity/animal/IronGolem.java
index 8885ab5..009e0ae 100644
--- a/net/minecraft/world/entity/animal/IronGolem.java
+++ b/net/minecraft/world/entity/animal/IronGolem.java
@@ -95,8 +95,8 @@ public class IronGolem extends AbstractGolem implements NeutralMob {
     // Purpur start - Configurable entity base attributes
     @Override
     public void initAttributes() {
-        this.getAttribute(Attributes.MAX_HEALTH).setBaseValue(this.level().purpurConfig.ironGolemMaxHealth);
-        this.getAttribute(Attributes.SCALE).setBaseValue(this.level().purpurConfig.ironGolemScale);
+        this.setAttributeBaseValue(Attributes.MAX_HEALTH, this.level().purpurConfig.ironGolemMaxHealth);
+        this.setAttributeBaseValue(Attributes.SCALE, this.level().purpurConfig.ironGolemScale);
     }
     // Purpur end - Configurable entity base attributes
 
diff --git a/net/minecraft/world/entity/animal/MushroomCow.java b/net/minecraft/world/entity/animal/MushroomCow.java
index cdb11ea..86bdc7f 100644
--- a/net/minecraft/world/entity/animal/MushroomCow.java
+++ b/net/minecraft/world/entity/animal/MushroomCow.java
@@ -82,7 +82,7 @@ public class MushroomCow extends AbstractCow implements Shearable {
     // Purpur start - Configurable entity base attributes
     @Override
     public void initAttributes() {
-        this.getAttribute(net.minecraft.world.entity.ai.attributes.Attributes.MAX_HEALTH).setBaseValue(this.level().purpurConfig.mooshroomMaxHealth);
+        this.setAttributeBaseValue(net.minecraft.world.entity.ai.attributes.Attributes.MAX_HEALTH, this.level().purpurConfig.mooshroomMaxHealth);
     }
     // Purpur end - Configurable entity base attributes
 
diff --git a/net/minecraft/world/entity/animal/Ocelot.java b/net/minecraft/world/entity/animal/Ocelot.java
index 4c20569..c83d47c 100644
--- a/net/minecraft/world/entity/animal/Ocelot.java
+++ b/net/minecraft/world/entity/animal/Ocelot.java
@@ -84,8 +84,8 @@ public class Ocelot extends Animal {
     // Purpur start - Configurable entity base attributes
     @Override
     public void initAttributes() {
-        this.getAttribute(Attributes.MAX_HEALTH).setBaseValue(this.level().purpurConfig.ocelotMaxHealth);
-        this.getAttribute(Attributes.SCALE).setBaseValue(this.level().purpurConfig.ocelotScale);
+        this.setAttributeBaseValue(Attributes.MAX_HEALTH, this.level().purpurConfig.ocelotMaxHealth);
+        this.setAttributeBaseValue(Attributes.SCALE, this.level().purpurConfig.ocelotScale);
     }
     // Purpur end - Configurable entity base attributes
 
diff --git a/net/minecraft/world/entity/animal/Panda.java b/net/minecraft/world/entity/animal/Panda.java
index 02bfa88..c4cfcab 100644
--- a/net/minecraft/world/entity/animal/Panda.java
+++ b/net/minecraft/world/entity/animal/Panda.java
@@ -136,9 +136,8 @@ public class Panda extends Animal {
     // Purpur start - Configurable entity base attributes
     @Override
     public void initAttributes() {
-        this.getAttribute(Attributes.MAX_HEALTH).setBaseValue(this.level().purpurConfig.pandaMaxHealth);
-        this.getAttribute(Attributes.SCALE).setBaseValue(this.level().purpurConfig.pandaScale);
-        setAttributes();
+        this.setAttributeBaseValue(Attributes.MAX_HEALTH, this.level().purpurConfig.pandaMaxHealth);
+        this.setAttributeBaseValue(Attributes.SCALE, this.level().purpurConfig.pandaScale);
     }
     // Purpur end - Configurable entity base attributes
 
diff --git a/net/minecraft/world/entity/animal/Parrot.java b/net/minecraft/world/entity/animal/Parrot.java
index dddf3a2..4989ece 100644
--- a/net/minecraft/world/entity/animal/Parrot.java
+++ b/net/minecraft/world/entity/animal/Parrot.java
@@ -196,8 +196,8 @@ public class Parrot extends ShoulderRidingEntity implements FlyingAnimal {
     // Purpur start - Configurable entity base attributes
     @Override
     public void initAttributes() {
-        this.getAttribute(Attributes.MAX_HEALTH).setBaseValue(this.level().purpurConfig.parrotMaxHealth);
-        this.getAttribute(Attributes.SCALE).setBaseValue(this.level().purpurConfig.parrotScale);
+        this.setAttributeBaseValue(Attributes.MAX_HEALTH, this.level().purpurConfig.parrotMaxHealth);
+        this.setAttributeBaseValue(Attributes.SCALE, this.level().purpurConfig.parrotScale);
     }
     // Purpur end - Configurable entity base attributes
 
diff --git a/net/minecraft/world/entity/animal/Pig.java b/net/minecraft/world/entity/animal/Pig.java
index adbbe93..fbda6b3 100644
--- a/net/minecraft/world/entity/animal/Pig.java
+++ b/net/minecraft/world/entity/animal/Pig.java
@@ -86,8 +86,8 @@ public class Pig extends Animal implements ItemSteerable {
     // Purpur start - Configurable entity base attributes
     @Override
     public void initAttributes() {
-        this.getAttribute(Attributes.MAX_HEALTH).setBaseValue(this.level().purpurConfig.pigMaxHealth);
-        this.getAttribute(Attributes.SCALE).setBaseValue(this.level().purpurConfig.pigScale);
+        this.setAttributeBaseValue(Attributes.MAX_HEALTH, this.level().purpurConfig.pigMaxHealth);
+        this.setAttributeBaseValue(Attributes.SCALE, this.level().purpurConfig.pigScale);
     }
     // Purpur end - Configurable entity base attributes
 
diff --git a/net/minecraft/world/entity/animal/PolarBear.java b/net/minecraft/world/entity/animal/PolarBear.java
index e0b51b4..a8cbddd 100644
--- a/net/minecraft/world/entity/animal/PolarBear.java
+++ b/net/minecraft/world/entity/animal/PolarBear.java
@@ -120,8 +120,8 @@ public class PolarBear extends Animal implements NeutralMob {
     // Purpur start - Configurable entity base attributes
     @Override
     public void initAttributes() {
-        this.getAttribute(Attributes.MAX_HEALTH).setBaseValue(this.level().purpurConfig.polarBearMaxHealth);
-        this.getAttribute(Attributes.SCALE).setBaseValue(this.level().purpurConfig.polarBearScale);
+        this.setAttributeBaseValue(Attributes.MAX_HEALTH, this.level().purpurConfig.polarBearMaxHealth);
+        this.setAttributeBaseValue(Attributes.SCALE, this.level().purpurConfig.polarBearScale);
     }
     // Purpur end - Configurable entity base attributes
 
diff --git a/net/minecraft/world/entity/animal/Pufferfish.java b/net/minecraft/world/entity/animal/Pufferfish.java
index 685176c..7c6ccac 100644
--- a/net/minecraft/world/entity/animal/Pufferfish.java
+++ b/net/minecraft/world/entity/animal/Pufferfish.java
@@ -62,7 +62,7 @@ public class Pufferfish extends AbstractFish {
     // Purpur start - Configurable entity base attributes
     @Override
     public void initAttributes() {
-        this.getAttribute(net.minecraft.world.entity.ai.attributes.Attributes.MAX_HEALTH).setBaseValue(this.level().purpurConfig.pufferfishMaxHealth);
+        this.setAttributeBaseValue(net.minecraft.world.entity.ai.attributes.Attributes.MAX_HEALTH, this.level().purpurConfig.pufferfishMaxHealth);
     }
     // Purpur end - Configurable entity base attributes
 
diff --git a/net/minecraft/world/entity/animal/Rabbit.java b/net/minecraft/world/entity/animal/Rabbit.java
index 041ccb9..c0a63a8 100644
--- a/net/minecraft/world/entity/animal/Rabbit.java
+++ b/net/minecraft/world/entity/animal/Rabbit.java
@@ -148,8 +148,8 @@ public class Rabbit extends Animal {
     // Purpur start - Configurable entity base attributes
     @Override
     public void initAttributes() {
-        this.getAttribute(Attributes.MAX_HEALTH).setBaseValue(this.level().purpurConfig.rabbitMaxHealth);
-        this.getAttribute(Attributes.SCALE).setBaseValue(this.level().purpurConfig.rabbitScale);
+        this.setAttributeBaseValue(Attributes.MAX_HEALTH, this.level().purpurConfig.rabbitMaxHealth);
+        this.setAttributeBaseValue(Attributes.SCALE, this.level().purpurConfig.rabbitScale);
     }
     // Purpur end - Configurable entity base attributes
 
diff --git a/net/minecraft/world/entity/animal/Salmon.java b/net/minecraft/world/entity/animal/Salmon.java
index d2679be..3156a83 100644
--- a/net/minecraft/world/entity/animal/Salmon.java
+++ b/net/minecraft/world/entity/animal/Salmon.java
@@ -54,7 +54,7 @@ public class Salmon extends AbstractSchoolingFish {
     // Purpur start - Configurable entity base attributes
     @Override
     public void initAttributes() {
-        this.getAttribute(net.minecraft.world.entity.ai.attributes.Attributes.MAX_HEALTH).setBaseValue(this.level().purpurConfig.salmonMaxHealth);
+        this.setAttributeBaseValue(net.minecraft.world.entity.ai.attributes.Attributes.MAX_HEALTH, this.level().purpurConfig.salmonMaxHealth);
     }
     // Purpur end - Configurable entity base attributes
 
diff --git a/net/minecraft/world/entity/animal/SnowGolem.java b/net/minecraft/world/entity/animal/SnowGolem.java
index 8984a4f..f1ff330 100644
--- a/net/minecraft/world/entity/animal/SnowGolem.java
+++ b/net/minecraft/world/entity/animal/SnowGolem.java
@@ -83,8 +83,8 @@ public class SnowGolem extends AbstractGolem implements Shearable, RangedAttackM
     // Purpur start - Configurable entity base attributes
     @Override
     public void initAttributes() {
-        this.getAttribute(Attributes.MAX_HEALTH).setBaseValue(this.level().purpurConfig.snowGolemMaxHealth);
-        this.getAttribute(Attributes.SCALE).setBaseValue(this.level().purpurConfig.snowGolemScale);
+        this.setAttributeBaseValue(Attributes.MAX_HEALTH, this.level().purpurConfig.snowGolemMaxHealth);
+        this.setAttributeBaseValue(Attributes.SCALE, this.level().purpurConfig.snowGolemScale);
     }
     // Purpur end - Configurable entity base attributes
 
diff --git a/net/minecraft/world/entity/animal/Squid.java b/net/minecraft/world/entity/animal/Squid.java
index 8371f38..0e4c460 100644
--- a/net/minecraft/world/entity/animal/Squid.java
+++ b/net/minecraft/world/entity/animal/Squid.java
@@ -94,8 +94,8 @@ public class Squid extends AgeableWaterCreature {
     // Purpur start - Configurable entity base attributes
     @Override
     public void initAttributes() {
-        this.getAttribute(Attributes.MAX_HEALTH).setBaseValue(this.level().purpurConfig.squidMaxHealth);
-        this.getAttribute(Attributes.SCALE).setBaseValue(this.level().purpurConfig.squidScale);
+        this.setAttributeBaseValue(Attributes.MAX_HEALTH, this.level().purpurConfig.squidMaxHealth);
+        this.setAttributeBaseValue(Attributes.SCALE, this.level().purpurConfig.squidScale);
     }
     // Purpur end - Configurable entity base attributes
 
diff --git a/net/minecraft/world/entity/animal/TropicalFish.java b/net/minecraft/world/entity/animal/TropicalFish.java
index 8d1fa86..35d6f1f 100644
--- a/net/minecraft/world/entity/animal/TropicalFish.java
+++ b/net/minecraft/world/entity/animal/TropicalFish.java
@@ -92,7 +92,7 @@ public class TropicalFish extends AbstractSchoolingFish {
     // Purpur start - Configurable entity base attributes
     @Override
     public void initAttributes() {
-        this.getAttribute(net.minecraft.world.entity.ai.attributes.Attributes.MAX_HEALTH).setBaseValue(this.level().purpurConfig.tropicalFishMaxHealth);
+        this.setAttributeBaseValue(net.minecraft.world.entity.ai.attributes.Attributes.MAX_HEALTH, this.level().purpurConfig.tropicalFishMaxHealth);
     }
     // Purpur end - Configurable entity base attributes
 
diff --git a/net/minecraft/world/entity/animal/Turtle.java b/net/minecraft/world/entity/animal/Turtle.java
index 9bdc1f1..0065a43 100644
--- a/net/minecraft/world/entity/animal/Turtle.java
+++ b/net/minecraft/world/entity/animal/Turtle.java
@@ -106,8 +106,8 @@ public class Turtle extends Animal {
     // Purpur start - Configurable entity base attributes
     @Override
     public void initAttributes() {
-        this.getAttribute(Attributes.MAX_HEALTH).setBaseValue(this.level().purpurConfig.turtleMaxHealth);
-        this.getAttribute(Attributes.SCALE).setBaseValue(this.level().purpurConfig.turtleScale);
+        this.setAttributeBaseValue(Attributes.MAX_HEALTH, this.level().purpurConfig.turtleMaxHealth);
+        this.setAttributeBaseValue(Attributes.SCALE, this.level().purpurConfig.turtleScale);
     }
     // Purpur end - Configurable entity base attributes
 
diff --git a/net/minecraft/world/entity/animal/allay/Allay.java b/net/minecraft/world/entity/animal/allay/Allay.java
index 0291f80..294f547 100644
--- a/net/minecraft/world/entity/animal/allay/Allay.java
+++ b/net/minecraft/world/entity/animal/allay/Allay.java
@@ -169,8 +169,8 @@ public class Allay extends PathfinderMob implements InventoryCarrier, VibrationS
     // Purpur start - Configurable entity base attributes
     @Override
     public void initAttributes() {
-        this.getAttribute(Attributes.MAX_HEALTH).setBaseValue(this.level().purpurConfig.allayMaxHealth);
-        this.getAttribute(Attributes.SCALE).setBaseValue(this.level().purpurConfig.allayScale);
+        this.setAttributeBaseValue(Attributes.MAX_HEALTH, this.level().purpurConfig.allayMaxHealth);
+        this.setAttributeBaseValue(Attributes.SCALE, this.level().purpurConfig.allayScale);
     }
     // Purpur end - Configurable entity base attributes
 
diff --git a/net/minecraft/world/entity/animal/armadillo/Armadillo.java b/net/minecraft/world/entity/animal/armadillo/Armadillo.java
index c520c4e..1a0ade8 100644
--- a/net/minecraft/world/entity/animal/armadillo/Armadillo.java
+++ b/net/minecraft/world/entity/animal/armadillo/Armadillo.java
@@ -100,8 +100,8 @@ public class Armadillo extends Animal {
     // Purpur start - Configurable entity base attributes
     @Override
     public void initAttributes() {
-        this.getAttribute(Attributes.MAX_HEALTH).setBaseValue(this.level().purpurConfig.armadilloMaxHealth);
-        this.getAttribute(Attributes.SCALE).setBaseValue(this.level().purpurConfig.armadilloScale);
+        this.setAttributeBaseValue(Attributes.MAX_HEALTH, this.level().purpurConfig.armadilloMaxHealth);
+        this.setAttributeBaseValue(Attributes.SCALE, this.level().purpurConfig.armadilloScale);
     }
     // Purpur end - Configurable entity base attributes
 
diff --git a/net/minecraft/world/entity/animal/axolotl/Axolotl.java b/net/minecraft/world/entity/animal/axolotl/Axolotl.java
index d8a65c8..6441f2c 100644
--- a/net/minecraft/world/entity/animal/axolotl/Axolotl.java
+++ b/net/minecraft/world/entity/animal/axolotl/Axolotl.java
@@ -140,8 +140,8 @@ public class Axolotl extends Animal implements Bucketable {
     // Purpur start - Configurable entity base attributes
     @Override
     public void initAttributes() {
-        this.getAttribute(Attributes.MAX_HEALTH).setBaseValue(this.level().purpurConfig.axolotlMaxHealth);
-        this.getAttribute(Attributes.SCALE).setBaseValue(this.level().purpurConfig.axolotlScale);
+        this.setAttributeBaseValue(Attributes.MAX_HEALTH, this.level().purpurConfig.axolotlMaxHealth);
+        this.setAttributeBaseValue(Attributes.SCALE, this.level().purpurConfig.axolotlScale);
     }
     // Purpur end - Configurable entity base attributes
 
diff --git a/net/minecraft/world/entity/animal/frog/Frog.java b/net/minecraft/world/entity/animal/frog/Frog.java
index d337b7f..2a40fc8 100644
--- a/net/minecraft/world/entity/animal/frog/Frog.java
+++ b/net/minecraft/world/entity/animal/frog/Frog.java
@@ -183,7 +183,7 @@ public class Frog extends Animal {
 
     @Override
     public void initAttributes() {
-        this.getAttribute(Attributes.MAX_HEALTH).setBaseValue(level().purpurConfig.frogMaxHealth);
+        this.setAttributeBaseValue(Attributes.MAX_HEALTH, this.level().purpurConfig.frogMaxHealth);
     }
     // Leaf end - Plazma - Add missing purpur configuration options
 
diff --git a/net/minecraft/world/entity/animal/frog/Tadpole.java b/net/minecraft/world/entity/animal/frog/Tadpole.java
index 8ede266..c4393cc 100644
--- a/net/minecraft/world/entity/animal/frog/Tadpole.java
+++ b/net/minecraft/world/entity/animal/frog/Tadpole.java
@@ -121,7 +121,7 @@ public class Tadpole extends AbstractFish {
 
     @Override
     public void initAttributes() {
-        this.getAttribute(Attributes.MAX_HEALTH).setBaseValue(level().purpurConfig.tadpoleMaxHealth);
+        this.setAttributeBaseValue(Attributes.MAX_HEALTH, this.level().purpurConfig.tadpoleMaxHealth);
     }
     // Leaf end - Plazma - Add missing purpur configuration options
 
diff --git a/net/minecraft/world/entity/animal/horse/AbstractHorse.java b/net/minecraft/world/entity/animal/horse/AbstractHorse.java
index 6f10062..d3b21d7 100644
--- a/net/minecraft/world/entity/animal/horse/AbstractHorse.java
+++ b/net/minecraft/world/entity/animal/horse/AbstractHorse.java
@@ -142,9 +142,9 @@ public abstract class AbstractHorse extends Animal implements HasCustomInventory
     // Purpur start - Configurable entity base attributes
     @Override
     public void initAttributes() {
-        this.getAttribute(Attributes.MAX_HEALTH).setBaseValue(this.generateMaxHealth(random));
-        this.getAttribute(Attributes.MOVEMENT_SPEED).setBaseValue(this.generateSpeed(random));
-        this.getAttribute(Attributes.JUMP_STRENGTH).setBaseValue(this.generateJumpStrength(random));
+        this.setAttributeBaseValue(Attributes.MAX_HEALTH, this.generateMaxHealth(random));
+        this.setAttributeBaseValue(Attributes.MOVEMENT_SPEED, this.generateSpeed(random));
+        this.setAttributeBaseValue(Attributes.JUMP_STRENGTH, this.generateJumpStrength(random));
     }
 
     protected double generateMaxHealth(double min, double max) {
diff --git a/net/minecraft/world/entity/animal/sheep/Sheep.java b/net/minecraft/world/entity/animal/sheep/Sheep.java
index 500c87f..d0b40c1 100644
--- a/net/minecraft/world/entity/animal/sheep/Sheep.java
+++ b/net/minecraft/world/entity/animal/sheep/Sheep.java
@@ -82,8 +82,8 @@ public class Sheep extends Animal implements Shearable {
     // Purpur start - Configurable entity base attributes
     @Override
     public void initAttributes() {
-        this.getAttribute(Attributes.MAX_HEALTH).setBaseValue(this.level().purpurConfig.sheepMaxHealth);
-        this.getAttribute(Attributes.SCALE).setBaseValue(this.level().purpurConfig.sheepScale);
+        this.setAttributeBaseValue(Attributes.MAX_HEALTH, this.level().purpurConfig.sheepMaxHealth);
+        this.setAttributeBaseValue(Attributes.SCALE, this.level().purpurConfig.sheepScale);
     }
     // Purpur end - Configurable entity base attributes
 
diff --git a/net/minecraft/world/entity/animal/sniffer/Sniffer.java b/net/minecraft/world/entity/animal/sniffer/Sniffer.java
index 4d872b3..01ddbed 100644
--- a/net/minecraft/world/entity/animal/sniffer/Sniffer.java
+++ b/net/minecraft/world/entity/animal/sniffer/Sniffer.java
@@ -107,8 +107,8 @@ public class Sniffer extends Animal {
     // Purpur start - Configurable entity base attributes
     @Override
     public void initAttributes() {
-        this.getAttribute(Attributes.MAX_HEALTH).setBaseValue(this.level().purpurConfig.snifferMaxHealth);
-        this.getAttribute(Attributes.SCALE).setBaseValue(this.level().purpurConfig.snifferScale);
+        this.setAttributeBaseValue(Attributes.MAX_HEALTH, this.level().purpurConfig.snifferMaxHealth);
+        this.setAttributeBaseValue(Attributes.SCALE, this.level().purpurConfig.snifferScale);
     }
     // Purpur end - Configurable entity base attributes
 
diff --git a/net/minecraft/world/entity/animal/wolf/Wolf.java b/net/minecraft/world/entity/animal/wolf/Wolf.java
index 7e7cb9d..5d3fd6a 100644
--- a/net/minecraft/world/entity/animal/wolf/Wolf.java
+++ b/net/minecraft/world/entity/animal/wolf/Wolf.java
@@ -211,8 +211,8 @@ public class Wolf extends TamableAnimal implements NeutralMob {
     // Purpur start - Configurable entity base attributes
     @Override
     public void initAttributes() {
-        this.getAttribute(Attributes.MAX_HEALTH).setBaseValue(this.level().purpurConfig.wolfMaxHealth);
-        this.getAttribute(Attributes.SCALE).setBaseValue(this.level().purpurConfig.wolfScale);
+        this.setAttributeBaseValue(Attributes.MAX_HEALTH, this.level().purpurConfig.wolfMaxHealth);
+        this.setAttributeBaseValue(Attributes.SCALE, this.level().purpurConfig.wolfScale);
     }
     // Purpur end - Configurable entity base attributes
 
diff --git a/net/minecraft/world/entity/boss/enderdragon/EnderDragon.java b/net/minecraft/world/entity/boss/enderdragon/EnderDragon.java
index e3cdff5..e8b06d2 100644
--- a/net/minecraft/world/entity/boss/enderdragon/EnderDragon.java
+++ b/net/minecraft/world/entity/boss/enderdragon/EnderDragon.java
@@ -168,7 +168,7 @@ public class EnderDragon extends Mob implements Enemy {
     // Purpur start - Configurable entity base attributes
     @Override
     public void initAttributes() {
-        this.getAttribute(Attributes.MAX_HEALTH).setBaseValue(this.level().purpurConfig.enderDragonMaxHealth);
+        this.setAttributeBaseValue(Attributes.MAX_HEALTH, this.level().purpurConfig.enderDragonMaxHealth);
     }
     // Purpur end - Configurable entity base attributes
 
diff --git a/net/minecraft/world/entity/boss/wither/WitherBoss.java b/net/minecraft/world/entity/boss/wither/WitherBoss.java
index 3171868..164716c 100644
--- a/net/minecraft/world/entity/boss/wither/WitherBoss.java
+++ b/net/minecraft/world/entity/boss/wither/WitherBoss.java
@@ -216,8 +216,8 @@ public class WitherBoss extends Monster implements RangedAttackMob {
     // Purpur start - Configurable entity base attributes
     @Override
     public void initAttributes() {
-        this.getAttribute(Attributes.MAX_HEALTH).setBaseValue(this.level().purpurConfig.witherMaxHealth);
-        this.getAttribute(Attributes.SCALE).setBaseValue(this.level().purpurConfig.witherScale);
+        this.setAttributeBaseValue(Attributes.MAX_HEALTH, this.level().purpurConfig.witherMaxHealth);
+        this.setAttributeBaseValue(Attributes.SCALE, this.level().purpurConfig.witherScale);
     }
     // Purpur end - Configurable entity base attributes
 
diff --git a/net/minecraft/world/entity/monster/Blaze.java b/net/minecraft/world/entity/monster/Blaze.java
index fa21b7d..032be48 100644
--- a/net/minecraft/world/entity/monster/Blaze.java
+++ b/net/minecraft/world/entity/monster/Blaze.java
@@ -78,8 +78,8 @@ public class Blaze extends Monster {
     // Purpur start - Configurable entity base attributes
     @Override
     public void initAttributes() {
-        this.getAttribute(Attributes.MAX_HEALTH).setBaseValue(this.level().purpurConfig.blazeMaxHealth);
-        this.getAttribute(Attributes.SCALE).setBaseValue(this.level().purpurConfig.blazeScale);
+        this.setAttributeBaseValue(Attributes.MAX_HEALTH, this.level().purpurConfig.blazeMaxHealth);
+        this.setAttributeBaseValue(Attributes.SCALE, this.level().purpurConfig.blazeScale);
     }
     // Purpur end - Configurable entity base attributes
 
diff --git a/net/minecraft/world/entity/monster/Bogged.java b/net/minecraft/world/entity/monster/Bogged.java
index cace310..f9e6d9a 100644
--- a/net/minecraft/world/entity/monster/Bogged.java
+++ b/net/minecraft/world/entity/monster/Bogged.java
@@ -63,8 +63,8 @@ public class Bogged extends AbstractSkeleton implements Shearable {
     // Purpur start - Configurable entity base attributes
     @Override
     public void initAttributes() {
-        this.getAttribute(Attributes.MAX_HEALTH).setBaseValue(this.level().purpurConfig.boggedMaxHealth);
-        this.getAttribute(Attributes.SCALE).setBaseValue(this.level().purpurConfig.boggedScale);
+        this.setAttributeBaseValue(Attributes.MAX_HEALTH, this.level().purpurConfig.boggedMaxHealth);
+        this.setAttributeBaseValue(Attributes.SCALE, this.level().purpurConfig.boggedScale);
     }
     // Purpur end - Configurable entity base attributes
 
diff --git a/net/minecraft/world/entity/monster/CaveSpider.java b/net/minecraft/world/entity/monster/CaveSpider.java
index 7eca4b7..2ba24d2 100644
--- a/net/minecraft/world/entity/monster/CaveSpider.java
+++ b/net/minecraft/world/entity/monster/CaveSpider.java
@@ -46,8 +46,8 @@ public class CaveSpider extends Spider {
     // Purpur start - Configurable entity base attributes
     @Override
     public void initAttributes() {
-        this.getAttribute(Attributes.MAX_HEALTH).setBaseValue(this.level().purpurConfig.caveSpiderMaxHealth);
-        this.getAttribute(Attributes.SCALE).setBaseValue(this.level().purpurConfig.caveSpiderScale);
+        this.setAttributeBaseValue(Attributes.MAX_HEALTH, this.level().purpurConfig.caveSpiderMaxHealth);
+        this.setAttributeBaseValue(Attributes.SCALE, this.level().purpurConfig.caveSpiderScale);
     }
     // Purpur end - Configurable entity base attributes
 
diff --git a/net/minecraft/world/entity/monster/Creeper.java b/net/minecraft/world/entity/monster/Creeper.java
index 2839c75..9bcfd34 100644
--- a/net/minecraft/world/entity/monster/Creeper.java
+++ b/net/minecraft/world/entity/monster/Creeper.java
@@ -139,8 +139,8 @@ public class Creeper extends Monster {
     // Purpur start - Configurable entity base attributes
     @Override
     public void initAttributes() {
-        this.getAttribute(Attributes.MAX_HEALTH).setBaseValue(this.level().purpurConfig.creeperMaxHealth);
-        this.getAttribute(Attributes.SCALE).setBaseValue(this.level().purpurConfig.creeperScale);
+        this.setAttributeBaseValue(Attributes.MAX_HEALTH, this.level().purpurConfig.creeperMaxHealth);
+        this.setAttributeBaseValue(Attributes.SCALE, this.level().purpurConfig.creeperScale);
     }
     // Purpur end - Configurable entity base attributes
 
diff --git a/net/minecraft/world/entity/monster/Drowned.java b/net/minecraft/world/entity/monster/Drowned.java
index 7ff3802..85c095c 100644
--- a/net/minecraft/world/entity/monster/Drowned.java
+++ b/net/minecraft/world/entity/monster/Drowned.java
@@ -96,9 +96,9 @@ public class Drowned extends Zombie implements RangedAttackMob {
     // Purpur start - Configurable entity base attributes
     @Override
     public void initAttributes() {
-        this.getAttribute(Attributes.MAX_HEALTH).setBaseValue(this.level().purpurConfig.drownedMaxHealth);
-        this.getAttribute(Attributes.SCALE).setBaseValue(this.level().purpurConfig.drownedScale);
-        this.getAttribute(Attributes.MOVEMENT_SPEED).setBaseValue(this.level().purpurConfig.drownedMovementSpeed); // Leaf - Configurable movement speed of more entities - drowned
+        this.setAttributeBaseValue(Attributes.MAX_HEALTH, this.level().purpurConfig.drownedMaxHealth);
+        this.setAttributeBaseValue(Attributes.SCALE, this.level().purpurConfig.drownedScale);
+        this.setAttributeBaseValue(Attributes.MOVEMENT_SPEED, this.level().purpurConfig.drownedMovementSpeed); // Leaf - Configurable movement speed of more entities - drowned
     }
 
     @Override
diff --git a/net/minecraft/world/entity/monster/ElderGuardian.java b/net/minecraft/world/entity/monster/ElderGuardian.java
index ab1c050..7207e98 100644
--- a/net/minecraft/world/entity/monster/ElderGuardian.java
+++ b/net/minecraft/world/entity/monster/ElderGuardian.java
@@ -46,8 +46,8 @@ public class ElderGuardian extends Guardian {
     // Purpur start - Configurable entity base attributes
     @Override
     public void initAttributes() {
-        this.getAttribute(Attributes.MAX_HEALTH).setBaseValue(this.level().purpurConfig.elderGuardianMaxHealth);
-        this.getAttribute(Attributes.SCALE).setBaseValue(this.level().purpurConfig.elderGuardianScale);
+        this.setAttributeBaseValue(Attributes.MAX_HEALTH, this.level().purpurConfig.elderGuardianMaxHealth);
+        this.setAttributeBaseValue(Attributes.SCALE, this.level().purpurConfig.elderGuardianScale);
     }
     // Purpur end - Configurable entity base attributes
 
diff --git a/net/minecraft/world/entity/monster/EnderMan.java b/net/minecraft/world/entity/monster/EnderMan.java
index 58887b2..301c760 100644
--- a/net/minecraft/world/entity/monster/EnderMan.java
+++ b/net/minecraft/world/entity/monster/EnderMan.java
@@ -109,8 +109,8 @@ public class EnderMan extends Monster implements NeutralMob {
     // Purpur start - Configurable entity base attributes
     @Override
     public void initAttributes() {
-        this.getAttribute(Attributes.MAX_HEALTH).setBaseValue(this.level().purpurConfig.endermanMaxHealth);
-        this.getAttribute(Attributes.SCALE).setBaseValue(this.level().purpurConfig.endermanScale);
+        this.setAttributeBaseValue(Attributes.MAX_HEALTH, this.level().purpurConfig.endermanMaxHealth);
+        this.setAttributeBaseValue(Attributes.SCALE, this.level().purpurConfig.endermanScale);
     }
     // Purpur end - Configurable entity base attributes
 
diff --git a/net/minecraft/world/entity/monster/Endermite.java b/net/minecraft/world/entity/monster/Endermite.java
index 4ea0e5f..e7bd6bf 100644
--- a/net/minecraft/world/entity/monster/Endermite.java
+++ b/net/minecraft/world/entity/monster/Endermite.java
@@ -67,8 +67,8 @@ public class Endermite extends Monster {
     // Purpur start - Configurable entity base attributes
     @Override
     public void initAttributes() {
-        this.getAttribute(Attributes.MAX_HEALTH).setBaseValue(this.level().purpurConfig.endermiteMaxHealth);
-        this.getAttribute(Attributes.SCALE).setBaseValue(this.level().purpurConfig.endermiteScale);
+        this.setAttributeBaseValue(Attributes.MAX_HEALTH, this.level().purpurConfig.endermiteMaxHealth);
+        this.setAttributeBaseValue(Attributes.SCALE, this.level().purpurConfig.endermiteScale);
     }
     // Purpur end - Configurable entity base attributes
 
diff --git a/net/minecraft/world/entity/monster/Evoker.java b/net/minecraft/world/entity/monster/Evoker.java
index 5c1992a..1b2c471 100644
--- a/net/minecraft/world/entity/monster/Evoker.java
+++ b/net/minecraft/world/entity/monster/Evoker.java
@@ -68,8 +68,8 @@ public class Evoker extends SpellcasterIllager {
     // Purpur start - Configurable entity base attributes
     @Override
     public void initAttributes() {
-        this.getAttribute(Attributes.MAX_HEALTH).setBaseValue(this.level().purpurConfig.evokerMaxHealth);
-        this.getAttribute(Attributes.SCALE).setBaseValue(this.level().purpurConfig.evokerScale);
+        this.setAttributeBaseValue(Attributes.MAX_HEALTH, this.level().purpurConfig.evokerMaxHealth);
+        this.setAttributeBaseValue(Attributes.SCALE, this.level().purpurConfig.evokerScale);
     }
     // Purpur end - Configurable entity base attributes
 
diff --git a/net/minecraft/world/entity/monster/Ghast.java b/net/minecraft/world/entity/monster/Ghast.java
index 6ec8387..c249946 100644
--- a/net/minecraft/world/entity/monster/Ghast.java
+++ b/net/minecraft/world/entity/monster/Ghast.java
@@ -77,8 +77,8 @@ public class Ghast extends Mob implements Enemy {
     // Purpur start - Configurable entity base attributes
     @Override
     public void initAttributes() {
-        this.getAttribute(Attributes.MAX_HEALTH).setBaseValue(this.level().purpurConfig.ghastMaxHealth);
-        this.getAttribute(Attributes.SCALE).setBaseValue(this.level().purpurConfig.ghastScale);
+        this.setAttributeBaseValue(Attributes.MAX_HEALTH, this.level().purpurConfig.ghastMaxHealth);
+        this.setAttributeBaseValue(Attributes.SCALE, this.level().purpurConfig.ghastScale);
     }
     // Purpur end - Configurable entity base attributes
 
diff --git a/net/minecraft/world/entity/monster/Giant.java b/net/minecraft/world/entity/monster/Giant.java
index f4ec41a..96e42cd 100644
--- a/net/minecraft/world/entity/monster/Giant.java
+++ b/net/minecraft/world/entity/monster/Giant.java
@@ -55,10 +55,10 @@ public class Giant extends Monster {
     // Purpur start - Configurable entity base attributes
     @Override
     protected void initAttributes() {
-        this.getAttribute(Attributes.MAX_HEALTH).setBaseValue(this.level().purpurConfig.giantMaxHealth);
-        this.getAttribute(Attributes.SCALE).setBaseValue(this.level().purpurConfig.giantScale);
-        this.getAttribute(Attributes.MOVEMENT_SPEED).setBaseValue(this.level().purpurConfig.giantMovementSpeed);
-        this.getAttribute(Attributes.ATTACK_DAMAGE).setBaseValue(this.level().purpurConfig.giantAttackDamage);
+        this.setAttributeBaseValue(Attributes.MAX_HEALTH, this.level().purpurConfig.giantMaxHealth);
+        this.setAttributeBaseValue(Attributes.SCALE, this.level().purpurConfig.giantScale);
+        this.setAttributeBaseValue(Attributes.MOVEMENT_SPEED, this.level().purpurConfig.giantMovementSpeed);
+        this.setAttributeBaseValue(Attributes.ATTACK_DAMAGE, this.level().purpurConfig.giantAttackDamage);
     }
     // Purpur end - Configurable entity base attributes
 
diff --git a/net/minecraft/world/entity/monster/Guardian.java b/net/minecraft/world/entity/monster/Guardian.java
index 3ba8da3..1ee5a25 100644
--- a/net/minecraft/world/entity/monster/Guardian.java
+++ b/net/minecraft/world/entity/monster/Guardian.java
@@ -93,8 +93,8 @@ public class Guardian extends Monster {
     // Purpur start - Configurable entity base attributes
     @Override
     public void initAttributes() {
-        this.getAttribute(Attributes.MAX_HEALTH).setBaseValue(this.level().purpurConfig.guardianMaxHealth);
-        this.getAttribute(Attributes.SCALE).setBaseValue(this.level().purpurConfig.guardianScale);
+        this.setAttributeBaseValue(Attributes.MAX_HEALTH, this.level().purpurConfig.guardianMaxHealth);
+        this.setAttributeBaseValue(Attributes.SCALE, this.level().purpurConfig.guardianScale);
     }
     // Purpur end - Configurable entity base attributes
 
diff --git a/net/minecraft/world/entity/monster/Husk.java b/net/minecraft/world/entity/monster/Husk.java
index f4200bd..1e7f7bf 100644
--- a/net/minecraft/world/entity/monster/Husk.java
+++ b/net/minecraft/world/entity/monster/Husk.java
@@ -42,8 +42,8 @@ public class Husk extends Zombie {
     // Purpur start - Configurable entity base attributes
     @Override
     public void initAttributes() {
-        this.getAttribute(net.minecraft.world.entity.ai.attributes.Attributes.MAX_HEALTH).setBaseValue(this.level().purpurConfig.huskMaxHealth);
-        this.getAttribute(net.minecraft.world.entity.ai.attributes.Attributes.MOVEMENT_SPEED).setBaseValue(this.level().purpurConfig.huskMovementSpeed); // Leaf - Configurable movement speed of more entities - husk
+        this.setAttributeBaseValue(net.minecraft.world.entity.ai.attributes.Attributes.MAX_HEALTH, this.level().purpurConfig.huskMaxHealth);
+        this.setAttributeBaseValue(net.minecraft.world.entity.ai.attributes.Attributes.MOVEMENT_SPEED, this.level().purpurConfig.huskMovementSpeed); // Leaf - Configurable movement speed of more entities - husk
     }
 
     @Override
diff --git a/net/minecraft/world/entity/monster/Illusioner.java b/net/minecraft/world/entity/monster/Illusioner.java
index 2ea8c84..7fe4694 100644
--- a/net/minecraft/world/entity/monster/Illusioner.java
+++ b/net/minecraft/world/entity/monster/Illusioner.java
@@ -77,10 +77,10 @@ public class Illusioner extends SpellcasterIllager implements RangedAttackMob {
     // Purpur start - Configurable entity base attributes
     @Override
     protected void initAttributes() {
-        this.getAttribute(Attributes.MOVEMENT_SPEED).setBaseValue(this.level().purpurConfig.illusionerMovementSpeed);
-        this.getAttribute(Attributes.FOLLOW_RANGE).setBaseValue(this.level().purpurConfig.illusionerFollowRange);
-        this.getAttribute(Attributes.MAX_HEALTH).setBaseValue(this.level().purpurConfig.illusionerMaxHealth);
-        this.getAttribute(Attributes.SCALE).setBaseValue(this.level().purpurConfig.illusionerScale);
+        this.setAttributeBaseValue(Attributes.MOVEMENT_SPEED, this.level().purpurConfig.illusionerMovementSpeed);
+        this.setAttributeBaseValue(Attributes.FOLLOW_RANGE, this.level().purpurConfig.illusionerFollowRange);
+        this.setAttributeBaseValue(Attributes.MAX_HEALTH, this.level().purpurConfig.illusionerMaxHealth);
+        this.setAttributeBaseValue(Attributes.SCALE, this.level().purpurConfig.illusionerScale);
     }
     // Purpur end - Configurable entity base attributes
 
diff --git a/net/minecraft/world/entity/monster/Pillager.java b/net/minecraft/world/entity/monster/Pillager.java
index 2ed4734..451ae7b 100644
--- a/net/minecraft/world/entity/monster/Pillager.java
+++ b/net/minecraft/world/entity/monster/Pillager.java
@@ -84,8 +84,8 @@ public class Pillager extends AbstractIllager implements CrossbowAttackMob, Inve
     // Purpur start - Configurable entity base attributes
     @Override
     public void initAttributes() {
-        this.getAttribute(Attributes.MAX_HEALTH).setBaseValue(this.level().purpurConfig.pillagerMaxHealth);
-        this.getAttribute(Attributes.SCALE).setBaseValue(this.level().purpurConfig.pillagerScale);
+        this.setAttributeBaseValue(Attributes.MAX_HEALTH, this.level().purpurConfig.pillagerMaxHealth);
+        this.setAttributeBaseValue(Attributes.SCALE, this.level().purpurConfig.pillagerScale);
     }
     // Purpur end - Configurable entity base attributes
 
diff --git a/net/minecraft/world/entity/monster/Ravager.java b/net/minecraft/world/entity/monster/Ravager.java
index 2b85ced..ac13a83 100644
--- a/net/minecraft/world/entity/monster/Ravager.java
+++ b/net/minecraft/world/entity/monster/Ravager.java
@@ -96,8 +96,8 @@ public class Ravager extends Raider {
     // Purpur start - Configurable entity base attributes
     @Override
     public void initAttributes() {
-        this.getAttribute(Attributes.MAX_HEALTH).setBaseValue(this.level().purpurConfig.ravagerMaxHealth);
-        this.getAttribute(Attributes.SCALE).setBaseValue(this.level().purpurConfig.ravagerScale);
+        this.setAttributeBaseValue(Attributes.MAX_HEALTH, this.level().purpurConfig.ravagerMaxHealth);
+        this.setAttributeBaseValue(Attributes.SCALE, this.level().purpurConfig.ravagerScale);
     }
     // Purpur end - Configurable entity base attributes
 
diff --git a/net/minecraft/world/entity/monster/Shulker.java b/net/minecraft/world/entity/monster/Shulker.java
index a46ca09..9a0e634 100644
--- a/net/minecraft/world/entity/monster/Shulker.java
+++ b/net/minecraft/world/entity/monster/Shulker.java
@@ -129,8 +129,8 @@ public class Shulker extends AbstractGolem implements Enemy {
     // Purpur start - Configurable entity base attributes
     @Override
     public void initAttributes() {
-        this.getAttribute(Attributes.MAX_HEALTH).setBaseValue(this.level().purpurConfig.shulkerMaxHealth);
-        this.getAttribute(Attributes.SCALE).setBaseValue(this.level().purpurConfig.shulkerScale);
+        this.setAttributeBaseValue(Attributes.MAX_HEALTH, this.level().purpurConfig.shulkerMaxHealth);
+        this.setAttributeBaseValue(Attributes.SCALE, this.level().purpurConfig.shulkerScale);
     }
     // Purpur end - Configurable entity base attributes
 
diff --git a/net/minecraft/world/entity/monster/Silverfish.java b/net/minecraft/world/entity/monster/Silverfish.java
index bcae390..bec362d 100644
--- a/net/minecraft/world/entity/monster/Silverfish.java
+++ b/net/minecraft/world/entity/monster/Silverfish.java
@@ -59,10 +59,10 @@ public class Silverfish extends Monster {
     // Purpur start - Configurable entity base attributes
     @Override
     public void initAttributes() {
-        this.getAttribute(Attributes.MAX_HEALTH).setBaseValue(this.level().purpurConfig.silverfishMaxHealth);
-        this.getAttribute(Attributes.SCALE).setBaseValue(this.level().purpurConfig.silverfishScale);
-        this.getAttribute(Attributes.MOVEMENT_SPEED).setBaseValue(this.level().purpurConfig.silverfishMovementSpeed);
-        this.getAttribute(Attributes.ATTACK_DAMAGE).setBaseValue(this.level().purpurConfig.silverfishAttackDamage);
+        this.setAttributeBaseValue(Attributes.MAX_HEALTH, this.level().purpurConfig.silverfishMaxHealth);
+        this.setAttributeBaseValue(Attributes.SCALE, this.level().purpurConfig.silverfishScale);
+        this.setAttributeBaseValue(Attributes.MOVEMENT_SPEED, this.level().purpurConfig.silverfishMovementSpeed);
+        this.setAttributeBaseValue(Attributes.ATTACK_DAMAGE, this.level().purpurConfig.silverfishAttackDamage);
     }
     // Purpur end - Configurable entity base attributes
 
diff --git a/net/minecraft/world/entity/monster/Skeleton.java b/net/minecraft/world/entity/monster/Skeleton.java
index c6ab31f..7683c6a 100644
--- a/net/minecraft/world/entity/monster/Skeleton.java
+++ b/net/minecraft/world/entity/monster/Skeleton.java
@@ -47,7 +47,7 @@ public class Skeleton extends AbstractSkeleton {
     // Purpur start - Configurable entity base attributes
     @Override
     public void initAttributes() {
-        this.getAttribute(net.minecraft.world.entity.ai.attributes.Attributes.MAX_HEALTH).setBaseValue(this.level().purpurConfig.skeletonMaxHealth);
+        this.setAttributeBaseValue(net.minecraft.world.entity.ai.attributes.Attributes.MAX_HEALTH, this.level().purpurConfig.skeletonMaxHealth);
     }
     // Purpur end - Configurable entity base attributes
 
diff --git a/net/minecraft/world/entity/monster/Spider.java b/net/minecraft/world/entity/monster/Spider.java
index 7c50ed0..22eaf7c 100644
--- a/net/minecraft/world/entity/monster/Spider.java
+++ b/net/minecraft/world/entity/monster/Spider.java
@@ -70,8 +70,8 @@ public class Spider extends Monster {
     // Purpur start - Configurable entity base attributes
     @Override
     public void initAttributes() {
-        this.getAttribute(Attributes.MAX_HEALTH).setBaseValue(this.level().purpurConfig.spiderMaxHealth);
-        this.getAttribute(Attributes.SCALE).setBaseValue(this.level().purpurConfig.spiderScale);
+        this.setAttributeBaseValue(Attributes.MAX_HEALTH, this.level().purpurConfig.spiderMaxHealth);
+        this.setAttributeBaseValue(Attributes.SCALE, this.level().purpurConfig.spiderScale);
     }
     // Purpur end - Configurable entity base attributes
 
diff --git a/net/minecraft/world/entity/monster/Stray.java b/net/minecraft/world/entity/monster/Stray.java
index e69a5c5..cfbb89f 100644
--- a/net/minecraft/world/entity/monster/Stray.java
+++ b/net/minecraft/world/entity/monster/Stray.java
@@ -42,7 +42,7 @@ public class Stray extends AbstractSkeleton {
     // Purpur start - Configurable entity base attributes
     @Override
     public void initAttributes() {
-        this.getAttribute(net.minecraft.world.entity.ai.attributes.Attributes.MAX_HEALTH).setBaseValue(this.level().purpurConfig.strayMaxHealth);
+        this.setAttributeBaseValue(net.minecraft.world.entity.ai.attributes.Attributes.MAX_HEALTH, this.level().purpurConfig.strayMaxHealth);
     }
     // Purpur end - Configurable entity base attributes
 
diff --git a/net/minecraft/world/entity/monster/Strider.java b/net/minecraft/world/entity/monster/Strider.java
index 31cace2..4a4f3ed 100644
--- a/net/minecraft/world/entity/monster/Strider.java
+++ b/net/minecraft/world/entity/monster/Strider.java
@@ -112,8 +112,8 @@ public class Strider extends Animal implements ItemSteerable {
     // Purpur start - Configurable entity base attributes
     @Override
     public void initAttributes() {
-        this.getAttribute(Attributes.MAX_HEALTH).setBaseValue(this.level().purpurConfig.striderMaxHealth);
-        this.getAttribute(Attributes.SCALE).setBaseValue(this.level().purpurConfig.striderScale);
+        this.setAttributeBaseValue(Attributes.MAX_HEALTH, this.level().purpurConfig.striderMaxHealth);
+        this.setAttributeBaseValue(Attributes.SCALE, this.level().purpurConfig.striderScale);
     }
     // Purpur end - Configurable entity base attributes
 
diff --git a/net/minecraft/world/entity/monster/Vex.java b/net/minecraft/world/entity/monster/Vex.java
index ab320cf..d22216b 100644
--- a/net/minecraft/world/entity/monster/Vex.java
+++ b/net/minecraft/world/entity/monster/Vex.java
@@ -107,8 +107,8 @@ public class Vex extends Monster implements TraceableEntity {
     // Purpur start - Configurable entity base attributes
     @Override
     public void initAttributes() {
-        this.getAttribute(Attributes.MAX_HEALTH).setBaseValue(this.level().purpurConfig.vexMaxHealth);
-        this.getAttribute(Attributes.SCALE).setBaseValue(this.level().purpurConfig.vexScale);
+        this.setAttributeBaseValue(Attributes.MAX_HEALTH, this.level().purpurConfig.vexMaxHealth);
+        this.setAttributeBaseValue(Attributes.SCALE, this.level().purpurConfig.vexScale);
     }
     // Purpur end - Configurable entity base attributes
 
diff --git a/net/minecraft/world/entity/monster/Vindicator.java b/net/minecraft/world/entity/monster/Vindicator.java
index 7f70b23..52b2f4e 100644
--- a/net/minecraft/world/entity/monster/Vindicator.java
+++ b/net/minecraft/world/entity/monster/Vindicator.java
@@ -76,8 +76,8 @@ public class Vindicator extends AbstractIllager {
     // Purpur start - Configurable entity base attributes
     @Override
     public void initAttributes() {
-        this.getAttribute(Attributes.MAX_HEALTH).setBaseValue(this.level().purpurConfig.vindicatorMaxHealth);
-        this.getAttribute(Attributes.SCALE).setBaseValue(this.level().purpurConfig.vindicatorScale);
+        this.setAttributeBaseValue(Attributes.MAX_HEALTH, this.level().purpurConfig.vindicatorMaxHealth);
+        this.setAttributeBaseValue(Attributes.SCALE, this.level().purpurConfig.vindicatorScale);
     }
     // Purpur end - Configurable entity base attributes
 
diff --git a/net/minecraft/world/entity/monster/Witch.java b/net/minecraft/world/entity/monster/Witch.java
index bfb0e33..5d0c0af 100644
--- a/net/minecraft/world/entity/monster/Witch.java
+++ b/net/minecraft/world/entity/monster/Witch.java
@@ -76,8 +76,8 @@ public class Witch extends Raider implements RangedAttackMob {
     // Purpur start - Configurable entity base attributes
     @Override
     public void initAttributes() {
-        this.getAttribute(Attributes.MAX_HEALTH).setBaseValue(this.level().purpurConfig.witchMaxHealth);
-        this.getAttribute(Attributes.SCALE).setBaseValue(this.level().purpurConfig.witchScale);
+        this.setAttributeBaseValue(Attributes.MAX_HEALTH, this.level().purpurConfig.witchMaxHealth);
+        this.setAttributeBaseValue(Attributes.SCALE, this.level().purpurConfig.witchScale);
     }
     // Purpur end - Configurable entity base attributes
 
diff --git a/net/minecraft/world/entity/monster/WitherSkeleton.java b/net/minecraft/world/entity/monster/WitherSkeleton.java
index b3f5b2e..1af64a3 100644
--- a/net/minecraft/world/entity/monster/WitherSkeleton.java
+++ b/net/minecraft/world/entity/monster/WitherSkeleton.java
@@ -54,8 +54,8 @@ public class WitherSkeleton extends AbstractSkeleton {
     // Purpur start - Configurable entity base attributes
     @Override
     public void initAttributes() {
-        this.getAttribute(Attributes.MAX_HEALTH).setBaseValue(this.level().purpurConfig.witherSkeletonMaxHealth);
-        this.getAttribute(Attributes.SCALE).setBaseValue(this.level().purpurConfig.witherSkeletonScale);
+        this.setAttributeBaseValue(Attributes.MAX_HEALTH, this.level().purpurConfig.witherSkeletonMaxHealth);
+        this.setAttributeBaseValue(Attributes.SCALE, this.level().purpurConfig.witherSkeletonScale);
     }
     // Purpur end - Configurable entity base attributes
 
diff --git a/net/minecraft/world/entity/monster/Zoglin.java b/net/minecraft/world/entity/monster/Zoglin.java
index 385e6d9..eb82a05 100644
--- a/net/minecraft/world/entity/monster/Zoglin.java
+++ b/net/minecraft/world/entity/monster/Zoglin.java
@@ -105,8 +105,8 @@ public class Zoglin extends Monster implements HoglinBase {
     // Purpur start - Configurable entity base attributes
     @Override
     public void initAttributes() {
-        this.getAttribute(Attributes.MAX_HEALTH).setBaseValue(this.level().purpurConfig.zoglinMaxHealth);
-        this.getAttribute(Attributes.SCALE).setBaseValue(this.level().purpurConfig.zoglinScale);
+        this.setAttributeBaseValue(Attributes.MAX_HEALTH, this.level().purpurConfig.zoglinMaxHealth);
+        this.setAttributeBaseValue(Attributes.SCALE, this.level().purpurConfig.zoglinScale);
     }
     // Purpur end - Configurable entity base attributes
 
diff --git a/net/minecraft/world/entity/monster/Zombie.java b/net/minecraft/world/entity/monster/Zombie.java
index c27aa33..8a8b473 100644
--- a/net/minecraft/world/entity/monster/Zombie.java
+++ b/net/minecraft/world/entity/monster/Zombie.java
@@ -124,9 +124,9 @@ public class Zombie extends Monster {
     // Purpur start - Configurable entity base attributes
     @Override
     public void initAttributes() {
-        this.getAttribute(Attributes.MAX_HEALTH).setBaseValue(this.level().purpurConfig.zombieMaxHealth);
-        this.getAttribute(Attributes.SCALE).setBaseValue(this.level().purpurConfig.zombieScale);
-        this.getAttribute(Attributes.MOVEMENT_SPEED).setBaseValue(this.level().purpurConfig.zombieMovementSpeed); // Leaf - Configurable movement speed of more entities - zombie
+        this.setAttributeBaseValue(Attributes.MAX_HEALTH, this.level().purpurConfig.zombieMaxHealth);
+        this.setAttributeBaseValue(Attributes.SCALE, this.level().purpurConfig.zombieScale);
+        this.setAttributeBaseValue(Attributes.MOVEMENT_SPEED, this.level().purpurConfig.zombieMovementSpeed); // Leaf - Configurable movement speed of more entities - zombie
     }
     // Purpur end - Configurable entity base attributes
 
diff --git a/net/minecraft/world/entity/monster/ZombieVillager.java b/net/minecraft/world/entity/monster/ZombieVillager.java
index e89e4c2..ced3734 100644
--- a/net/minecraft/world/entity/monster/ZombieVillager.java
+++ b/net/minecraft/world/entity/monster/ZombieVillager.java
@@ -96,8 +96,8 @@ public class ZombieVillager extends Zombie implements VillagerDataHolder {
     // Purpur start - Configurable entity base attributes
     @Override
     public void initAttributes() {
-        this.getAttribute(net.minecraft.world.entity.ai.attributes.Attributes.MAX_HEALTH).setBaseValue(this.level().purpurConfig.zombieVillagerMaxHealth);
-        this.getAttribute(net.minecraft.world.entity.ai.attributes.Attributes.MOVEMENT_SPEED).setBaseValue(this.level().purpurConfig.zombieVillagerMovementSpeed); // Leaf - Configurable movement speed of more entities - zombieVillager
+        this.setAttributeBaseValue(net.minecraft.world.entity.ai.attributes.Attributes.MAX_HEALTH, this.level().purpurConfig.zombieVillagerMaxHealth);
+        this.setAttributeBaseValue(net.minecraft.world.entity.ai.attributes.Attributes.MOVEMENT_SPEED, this.level().purpurConfig.zombieVillagerMovementSpeed); // Leaf - Configurable movement speed of more entities - zombieVillager
     }
 
     @Override
diff --git a/net/minecraft/world/entity/monster/ZombifiedPiglin.java b/net/minecraft/world/entity/monster/ZombifiedPiglin.java
index cc34940..41cb3ad 100644
--- a/net/minecraft/world/entity/monster/ZombifiedPiglin.java
+++ b/net/minecraft/world/entity/monster/ZombifiedPiglin.java
@@ -84,9 +84,9 @@ public class ZombifiedPiglin extends Zombie implements NeutralMob {
     // Purpur start - Configurable entity base attributes
     @Override
     public void initAttributes() {
-        this.getAttribute(Attributes.MAX_HEALTH).setBaseValue(this.level().purpurConfig.zombifiedPiglinMaxHealth);
-        this.getAttribute(Attributes.SCALE).setBaseValue(this.level().purpurConfig.zombifiedPiglinScale);
-        this.getAttribute(Attributes.MOVEMENT_SPEED).setBaseValue(this.level().purpurConfig.zombifiedPiglinMovementSpeed); // Leaf - Configurable movement speed of more entities - zombifiedPiglin
+        this.setAttributeBaseValue(Attributes.MAX_HEALTH, this.level().purpurConfig.zombifiedPiglinMaxHealth);
+        this.setAttributeBaseValue(Attributes.SCALE, this.level().purpurConfig.zombifiedPiglinScale);
+        this.setAttributeBaseValue(Attributes.MOVEMENT_SPEED, this.level().purpurConfig.zombifiedPiglinMovementSpeed); // Leaf - Configurable movement speed of more entities - zombifiedPiglin
     }
     // Purpur end - Configurable entity base attributes
 
diff --git a/net/minecraft/world/entity/monster/creaking/Creaking.java b/net/minecraft/world/entity/monster/creaking/Creaking.java
index ef0522f..9b49a3e 100644
--- a/net/minecraft/world/entity/monster/creaking/Creaking.java
+++ b/net/minecraft/world/entity/monster/creaking/Creaking.java
@@ -127,8 +127,8 @@ public class Creaking extends Monster {
     // Purpur start - Configurable entity base attributes
     @Override
     public void initAttributes() {
-        this.getAttribute(Attributes.MAX_HEALTH).setBaseValue(this.level().purpurConfig.creakingMaxHealth);
-        this.getAttribute(Attributes.SCALE).setBaseValue(this.level().purpurConfig.creakingScale);
+        this.setAttributeBaseValue(Attributes.MAX_HEALTH, this.level().purpurConfig.creakingMaxHealth);
+        this.setAttributeBaseValue(Attributes.SCALE, this.level().purpurConfig.creakingScale);
     }
     // Purpur end - Configurable entity base attributes
 
diff --git a/net/minecraft/world/entity/monster/hoglin/Hoglin.java b/net/minecraft/world/entity/monster/hoglin/Hoglin.java
index e64ca3d..0086c98 100644
--- a/net/minecraft/world/entity/monster/hoglin/Hoglin.java
+++ b/net/minecraft/world/entity/monster/hoglin/Hoglin.java
@@ -114,8 +114,8 @@ public class Hoglin extends Animal implements Enemy, HoglinBase {
     // Purpur start - Configurable entity base attributes
     @Override
     public void initAttributes() {
-        this.getAttribute(Attributes.MAX_HEALTH).setBaseValue(this.level().purpurConfig.hoglinMaxHealth);
-        this.getAttribute(Attributes.SCALE).setBaseValue(this.level().purpurConfig.hoglinScale);
+        this.setAttributeBaseValue(Attributes.MAX_HEALTH, this.level().purpurConfig.hoglinMaxHealth);
+        this.setAttributeBaseValue(Attributes.SCALE, this.level().purpurConfig.hoglinScale);
     }
     // Purpur end - Configurable entity base attributes
 
diff --git a/net/minecraft/world/entity/monster/piglin/Piglin.java b/net/minecraft/world/entity/monster/piglin/Piglin.java
index 91c60e2..efb5042 100644
--- a/net/minecraft/world/entity/monster/piglin/Piglin.java
+++ b/net/minecraft/world/entity/monster/piglin/Piglin.java
@@ -155,8 +155,8 @@ public class Piglin extends AbstractPiglin implements CrossbowAttackMob, Invento
     // Purpur start - Configurable entity base attributes
     @Override
     public void initAttributes() {
-        this.getAttribute(Attributes.MAX_HEALTH).setBaseValue(this.level().purpurConfig.piglinMaxHealth);
-        this.getAttribute(Attributes.SCALE).setBaseValue(this.level().purpurConfig.piglinScale);
+        this.setAttributeBaseValue(Attributes.MAX_HEALTH, this.level().purpurConfig.piglinMaxHealth);
+        this.setAttributeBaseValue(Attributes.SCALE, this.level().purpurConfig.piglinScale);
     }
     // Purpur end - Configurable entity base attributes
 
diff --git a/net/minecraft/world/entity/monster/piglin/PiglinBrute.java b/net/minecraft/world/entity/monster/piglin/PiglinBrute.java
index c4eb58d..97f00d3 100644
--- a/net/minecraft/world/entity/monster/piglin/PiglinBrute.java
+++ b/net/minecraft/world/entity/monster/piglin/PiglinBrute.java
@@ -83,8 +83,8 @@ public class PiglinBrute extends AbstractPiglin {
     // Purpur start - Configurable entity base attributes
     @Override
     public void initAttributes() {
-        this.getAttribute(Attributes.MAX_HEALTH).setBaseValue(this.level().purpurConfig.piglinBruteMaxHealth);
-        this.getAttribute(Attributes.SCALE).setBaseValue(this.level().purpurConfig.piglinBruteScale);
+        this.setAttributeBaseValue(Attributes.MAX_HEALTH, this.level().purpurConfig.piglinBruteMaxHealth);
+        this.setAttributeBaseValue(Attributes.SCALE, this.level().purpurConfig.piglinBruteScale);
     }
     // Purpur end - Configurable entity base attributes
 
diff --git a/net/minecraft/world/entity/monster/warden/Warden.java b/net/minecraft/world/entity/monster/warden/Warden.java
index 51c5675..c348728 100644
--- a/net/minecraft/world/entity/monster/warden/Warden.java
+++ b/net/minecraft/world/entity/monster/warden/Warden.java
@@ -161,7 +161,7 @@ public class Warden extends Monster implements VibrationSystem {
 
     @Override
     public void initAttributes() {
-        this.getAttribute(Attributes.MAX_HEALTH).setBaseValue(level().purpurConfig.wardenMaxHealth);
+        this.setAttributeBaseValue(Attributes.MAX_HEALTH, this.level().purpurConfig.wardenMaxHealth);
     }
     // Leaf end - Plazma - Add missing purpur configuration options
 
diff --git a/net/minecraft/world/entity/npc/Villager.java b/net/minecraft/world/entity/npc/Villager.java
index dfc959e..8b19e06 100644
--- a/net/minecraft/world/entity/npc/Villager.java
+++ b/net/minecraft/world/entity/npc/Villager.java
@@ -275,9 +275,9 @@ public class Villager extends AbstractVillager implements ReputationEventHandler
     // Purpur start - Configurable entity base attributes
     @Override
     public void initAttributes() {
-        this.getAttribute(Attributes.MAX_HEALTH).setBaseValue(this.level().purpurConfig.villagerMaxHealth);
-        this.getAttribute(Attributes.SCALE).setBaseValue(this.level().purpurConfig.villagerScale);
-        this.getAttribute(Attributes.TEMPT_RANGE).setBaseValue(this.level().purpurConfig.villagerTemptRange); // Purpur - Villagers follow emerald blocks
+        this.setAttributeBaseValue(Attributes.MAX_HEALTH, this.level().purpurConfig.villagerMaxHealth);
+        this.setAttributeBaseValue(Attributes.SCALE, this.level().purpurConfig.villagerScale);
+        this.setAttributeBaseValue(Attributes.TEMPT_RANGE, this.level().purpurConfig.villagerTemptRange); // Purpur - Villagers follow emerald blocks
     }
     // Purpur end - Configurable entity base attributes
 
diff --git a/net/minecraft/world/entity/npc/WanderingTrader.java b/net/minecraft/world/entity/npc/WanderingTrader.java
index dc6b305..3876bd5 100644
--- a/net/minecraft/world/entity/npc/WanderingTrader.java
+++ b/net/minecraft/world/entity/npc/WanderingTrader.java
@@ -87,8 +87,8 @@ public class WanderingTrader extends net.minecraft.world.entity.npc.AbstractVill
     // Purpur start - Configurable entity base attributes
     @Override
     public void initAttributes() {
-        this.getAttribute(net.minecraft.world.entity.ai.attributes.Attributes.MAX_HEALTH).setBaseValue(this.level().purpurConfig.wanderingTraderMaxHealth);
-        this.getAttribute(net.minecraft.world.entity.ai.attributes.Attributes.TEMPT_RANGE).setBaseValue(this.level().purpurConfig.wanderingTraderTemptRange); // Purpur - Villagers follow emerald blocks
+        this.setAttributeBaseValue(net.minecraft.world.entity.ai.attributes.Attributes.MAX_HEALTH, this.level().purpurConfig.wanderingTraderMaxHealth);
+        this.setAttributeBaseValue(net.minecraft.world.entity.ai.attributes.Attributes.TEMPT_RANGE, this.level().purpurConfig.wanderingTraderTemptRange); // Purpur - Villagers follow emerald blocks
     }
     // Purpur end - Configurable entity base attributes
 
