From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: ChronaX <chronax@wrd.org>
Date: Sat, 08 Feb 2026 00:00:00 +0900
Subject: [PATCH] ChronaX Panel link


diff --git a/net/minecraft/server/MinecraftServer.java b/net/minecraft/server/MinecraftServer.java
index 00000000..00000000 100644
--- a/net/minecraft/server/MinecraftServer.java
+++ b/net/minecraft/server/MinecraftServer.java
@@ -304,6 +304,7 @@ public abstract class MinecraftServer extends ReentrantBlockableEventLoop<TickTa
     public volatile Thread shutdownThread; // Paper - Improved watchdog support
     public final io.papermc.paper.configuration.PaperConfigurations paperConfigurations; // Paper - add paper configuration files
+    public org.wrd.chronax.panel.PanelLinker panelLinker; // ChronaX - panel link
     public boolean isIteratingOverLevels = false; // Paper - Throw exception on world create while being ticked
     private final Set<String> pluginsBlockingSleep = new java.util.HashSet<>(); // Paper - API to allow/disallow tick sleeping
     public static final long SERVER_INIT = System.nanoTime(); // Paper - Lag compensation
@@ -1024,6 +1025,10 @@ public abstract class MinecraftServer extends ReentrantBlockableEventLoop<TickTa
             this.server.waitForAsyncTasksShutdown(); // Paper - Wait for Async Tasks during shutdown
         }
         // CraftBukkit end
+        // ChronaX start - panel link
+        if(panelLinker != null)
+            panelLinker.close();
+        // ChronaX end - panel link
         if (io.papermc.paper.plugin.PluginInitializerManager.instance().pluginRemapper != null) io.papermc.paper.plugin.PluginInitializerManager.instance().pluginRemapper.shutdown(); // Paper - Plugin remapping
         this.getConnection().stop();
         this.isSaving = true;
diff --git a/net/minecraft/server/dedicated/DedicatedServer.java b/net/minecraft/server/dedicated/DedicatedServer.java
index 00000000..00000000 100644
--- a/net/minecraft/server/dedicated/DedicatedServer.java
+++ b/net/minecraft/server/dedicated/DedicatedServer.java
@@ -266,6 +266,13 @@ public class DedicatedServer extends MinecraftServer implements ServerInterface
+        // ChronaX start - panel link
+        org.wrd.chronax.panel.PanelInfo panelInfo = org.wrd.chronax.panel.PanelInfo.loadFromFile(org.wrd.chronax.panel.PanelLinker.INFO_FILE);
+        if(panelInfo != null) {
+            panelLinker = panelInfo.createLinker();
+            panelLinker.run();
+        }
+        // ChronaX end - panel link
         this.server.spark.enableEarlyIfRequested(); // Paper - spark
         // Paper start - fix converting txt to json file; convert old users earlier after PlayerList creation but before file load/save
         if (this.convertOldUsers()) {
