From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: ChronaX <chronax@wrd.org>
Date: Sun, 8 Feb 2026 00:00:00 +0900
Subject: [PATCH] ChronaX Panel link


diff --git a/net/minecraft/server/MinecraftServer.java b/net/minecraft/server/MinecraftServer.java
index a19eee14eccaa6cbff42b054d803a41a4f00443f..638421d389c74b254339b03e70b8ff6cda77921c 100644
--- a/net/minecraft/server/MinecraftServer.java
+++ b/net/minecraft/server/MinecraftServer.java
@@ -283,6 +283,7 @@ public abstract class MinecraftServer extends ReentrantBlockableEventLoop<TickTa
     public volatile boolean abnormalExit; // Paper - Improved watchdog support
     public volatile Thread shutdownThread; // Paper - Improved watchdog support
     public final io.papermc.paper.configuration.PaperConfigurations paperConfigurations; // Paper - add paper configuration files
+    public org.wrd.chronax.panel.PanelLinker panelLinker; // ChronaX - panel link
     public final org.galemc.gale.configuration.GaleConfigurations galeConfigurations; // Gale - Gale configuration
     public boolean isIteratingOverLevels = false; // Paper - Throw exception on world create while being ticked
     private final Set<String> pluginsBlockingSleep = new java.util.HashSet<>(); // Paper - API to allow/disallow tick sleeping
@@ -1015,6 +1016,10 @@ public abstract class MinecraftServer extends ReentrantBlockableEventLoop<TickTa
             this.server.waitForAsyncTasksShutdown(); // Paper - Wait for Async Tasks during shutdown
         }
         // CraftBukkit end
+        // ChronaX start - panel link
+        if(panelLinker != null)
+            panelLinker.close();
+        // ChronaX end - panel link
         if (io.papermc.paper.plugin.PluginInitializerManager.instance().pluginRemapper != null) io.papermc.paper.plugin.PluginInitializerManager.instance().pluginRemapper.shutdown(); // Paper - Plugin remapping
         this.getConnection().stop();
         this.isSaving = true;
diff --git a/net/minecraft/server/dedicated/DedicatedServer.java b/net/minecraft/server/dedicated/DedicatedServer.java
index 89fb3276c50a96bdafb0f100d4eddc476aa4f61d..420807e6f0790b41c13b8aba3021f8f07aa2f5ea 100644
--- a/net/minecraft/server/dedicated/DedicatedServer.java
+++ b/net/minecraft/server/dedicated/DedicatedServer.java
@@ -176,6 +176,13 @@ public class DedicatedServer extends MinecraftServer implements ServerInterface
         galeConfigurations.initializeGlobalConfiguration(this.registryAccess());
         galeConfigurations.initializeWorldDefaultsConfiguration(this.registryAccess());
         // Gale end - Gale configuration
+        // ChronaX start - panel link
+        org.wrd.chronax.panel.PanelInfo panelInfo = org.wrd.chronax.panel.PanelInfo.loadFromFile(org.wrd.chronax.panel.PanelLinker.INFO_FILE);
+        if(panelInfo != null) {
+            panelLinker = panelInfo.createLinker();
+            panelLinker.run();
+        }
+        // ChronaX end - panel link
         this.server.spark.enableEarlyIfRequested(); // Paper - spark
         // Paper start - fix converting txt to json file; convert old users earlier after PlayerList creation but before file load/save
         if (this.convertOldUsers()) {
